prow/tide/tide.go

```diff
@@ -1547,7 +1548,7 @@ func (c *Controller) takeAction(sp subpool, batchPending, successes, pendings, m
 	}
 	// If we have no serial jobs pending or successful, trigger one.
 	if len(missings) > 0 && len(pendings) == 0 && len(successes) == 0 {
-		if ok, pr := pickHighestPriorityPR(sp.log, c.ghc, missings, sp.cc, isPassingTests, c.config().Tide.Priority); ok {
+		if ok, pr := pickHighestPriorityPR(sp.log, missings, sp.cc, c.isRetestEligible, c.config().Tide.Priority); ok {
```

Comment:
This is basically the relevant change, it now uses the same logic we already use in batches to determine if a job can be retested. Pretty much everything else is renames and signature changes (because `isPassingTests` and `isRetestEligible` need to have the same signature)